plutus-core/plutus-core/src/PlutusCore/ParserCommon.hs

```diff
@@ -0,0 +1,179 @@
+
+parseQuoted :: Parser a -> String -> T.Text -> PLC.Quote (Either (ParseErrorBundle T.Text ParseError) a)
+parseQuoted p file str = flip evalStateT initial $ runParserT p file str
+
+type Parser = ParsecT ParseError T.Text (StateT ParserState PLC.Quote)
+instance (Stream s, PLC.MonadQuote m) => PLC.MonadQuote (ParsecT e s m)
+
+topSourcePos :: SourcePos
+topSourcePos = initialPos "top"
+
+whitespace :: Parser ()
+whitespace = Lex.space space1 (Lex.skipLineComment "--") (Lex.skipBlockCommentNested "{-" "-}")
+
+lexeme :: Parser a -> Parser a
+lexeme = Lex.lexeme whitespace
+
+symbol :: T.Text -> Parser T.Text
+symbol = Lex.symbol whitespace
+
+lparen :: Parser T.Text
+lparen = symbol "("
+rparen :: Parser T.Text
+rparen = symbol ")"
+
+lbracket :: Parser T.Text
+lbracket = symbol "["
+rbracket :: Parser T.Text
+rbracket = symbol "]"
+
+lbrace :: Parser T.Text
+lbrace = symbol "{"
+rbrace :: Parser T.Text
+rbrace = symbol "}"
+
+inParens :: Parser a -> Parser a
+inParens = between lparen rparen
+
+inBrackets :: Parser a -> Parser a
+inBrackets = between lbracket rbracket
+
+inBraces :: Parser a -> Parser a
+inBraces = between lbrace rbrace
+
+isIdentifierChar :: Char -> Bool
+isIdentifierChar c = isAlphaNum c || c == '_' || c == '\''
+
+reservedWord :: T.Text -> Parser SourcePos
+reservedWord w = lexeme $ try $ do
+    p <- getSourcePos
+    void $ string w
+    notFollowedBy (satisfy isIdentifierChar)
+    return p
+
+-- | Parse a type tag by feeding the output of 'closedChunk' to 'PLC.parse'.
+builtinTypeTag
+    :: PLC.Parsable (PLC.Some uni)
+    => Parser (PLC.Some uni)
+builtinTypeTag = do
+    uniText <- closedChunk
+    case PLC.parse uniText of
+        Nothing  -> customFailure $ UnknownBuiltinType uniText
+        Just uni -> pure uni
+
+-- | Parse a constant by parsing a type tag first and using the type-specific parser of constants.
+-- Uses 'PLC.parse' under the hood for both types and constants.
+constant
+    :: (PLC.Parsable (PLC.Some uni), PLC.Closed uni, uni `PLC.Everywhere` PLC.Parsable)
+    => Parser (PLC.Some (PLC.ValueOf uni))
+constant = do
+    -- We use 'match' for remembering the textual representation of the parsed type tag,
+    -- so that we can show it in the error message if the constant fails to parse.
+    (uniText, PLC.Some uni) <- match builtinTypeTag
+    conText <- closedChunk
+    case PLC.bring (Proxy @PLC.Parsable) uni $ PLC.parse conText of
+        Nothing  -> customFailure $ InvalidConstant uniText conText
+        Just con -> pure . PLC.Some $ PLC.ValueOf uni con
+
+builtinFunction :: (Bounded fun, Enum fun, Pretty fun) => Parser fun
+builtinFunction = lexeme $ choice $ map parseBuiltin [minBound .. maxBound]
+    where parseBuiltin builtin = try $ string (display builtin) >> pure builtin
+
+name :: Parser PLC.Name
+name = lexeme $ try $ do
+    void $ lookAhead letterChar
+    str <- takeWhileP (Just "identifier") isIdentifierChar
+    PLC.Name str <$> intern str
+
+var :: Parser PLC.Name
```

Comment:
why not `name` and `tyname` for these parsers?

Reply:
I actually have no idea why the old parser has renamed `name`(L140) to `var` in L145. I'll delete L145-146 and rename `tyVar`.